File: Back End/Lnu_web/Services/ChatService/ChatService.cs
using Lnu_web.Data;
using Lnu_web.Dbo.Chat;
using Lnu_web.Interfaces.Services.IChatRepository;
using Lnu_web.Models.ChatModel;
using Lnu_web.Models.User;
using Lnu_web.Reposotories.ChatRepository;

namespace Lnu_web.Services.ChatService
{
    public class ChatService : IChatService
    {
        private readonly ChatRepository _chatRepository;

        public ChatService(ChatRepository chatRepository)
        {
            _chatRepository = chatRepository;
        }

        public async Task<List<ChatModel>> GetChatHistory(string senderId, string receiverId)
        {
            return await _chatRepository.GetChatHistory(senderId, receiverId);
        }
        public async Task<string> SaveMessage(ChatModel chatModel)
        {
            return await _chatRepository.SaveMessage(chatModel);
        }

        public async Task<List<ChatsListModel>> GetUserChats(string userId)
        {
            return await _chatRepository.GetUserChats(userId);
        }

        public async Task<bool> MakeMessageAsRead(string messageId)
        {
            return await _chatRepository.MakeMessageAsRead(messageId);
        }

        public async Task<string> CreateGroup(CreateGroupRequest groupRequest)
        {
            if (groupRequest != null)
            {
                return await _chatRepository.CreateGroup(groupRequest);
            }

            return null;
        }

        public async Task<string> SentGroupMessage(ChatModel model)
        {
            return await _chatRepository.SentGroupMessage(model);
        }

        public async Task<List<GroupMessageDbo>> GetGroupChatHistoryAsync(Guid groupId)
        {
            var messages = await _chatRepository.GetGroupChatHistoryAsync(groupId);

            if (messages != null)
            {
                return messages;
            };

            return null;
        }

        public async Task<bool> MakeGroupMessageAsRead(Guid messageId)
        {
            return await _chatRepository.MakeGroupMessageAsRead(messageId);
        }

        public async Task<List<UserModel>> GetUsersByGroup(Guid groupId)
        {
            if (groupId == Guid.Empty)
            {
                return null;
            }

            var usersList = await _chatRepository.GetUsersByGroup(groupId);

            if (usersList != null)
            {
                return usersList;
            }

            return null;
        }

        public async Task<bool> DeleteUserFromGroup(List<Guid> userIds, Guid groupId)
        {
            if (userIds == null || !userIds.Any())
                return false;

            return await _chatRepository.DeleteUserFromGroup(userIds, groupId);
        }

        public async Task<bool> AddUsersToGroup(AddUserRequest request)
        {
            return await _chatRepository.AddUsersToGroup(request);
        }
    }
}

File: Front End/LNU-Dekanat/src/app/core/services/chat/chat.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ChatModel } from '../../../models/chat/chat.module';

@Injectable({
  providedIn: 'root'
})

export class ChatService {

<<<<<<< HEAD
  private readonly baseUrl = 'http://localhost:7215/Chat'; 
=======
  private readonly baseUrl = 'http://localhost:5208/Chat'; 
>>>>>>> 875b81d (Initial commit to main)

  constructor(private http: HttpClient) {}

  sendMessage(model: ChatModel): Observable<any> {
    return this.http.post<any>(`${this.baseUrl}/sentToPrivateChat`, model);
  }

  getChatHistory(SenderId: string, ReceiverId: string): Observable<ChatModel[]> {
    const body = { SenderId, ReceiverId };
    return this.http.post<ChatModel[]>(`${this.baseUrl}/getHistory`, body);
  }

  getUserChats(userId: string): Observable<ChatModel[]> {
    return this.http.get<ChatModel[]>(`${this.baseUrl}/getUserChats${userId}`);
  }

  makeMessageAsRead(messageId: string) : Observable<any> {
    return this.http.patch<any>(`${this.baseUrl}/updateMessageReadStatus/${messageId}`, messageId)
  }

  createGroup(groupRequest: any) : Observable<any> {
    return this.http.post<any>(`${this.baseUrl}/createGroup`, groupRequest)
  }

  sentGroupMessage(model: ChatModel): Observable<any> {
    return this.http.post<any>(`${this.baseUrl}/sentGroupMessage`, model)
  }

  getGroupMessageHistory(groupId: string): Observable<ChatModel[]> {
    return this.http.get<ChatModel[]>(`${this.baseUrl}/getGroupChatHistory${groupId}`)
  }

  makeGroupMessageAsRead(messageId: string) : Observable<any> {
    return this.http.patch<any>(`${this.baseUrl}/updateGroupMessageReadStatus/${messageId}`, messageId)
  }

  getGroupUsers(groupId: string): Observable<any> {
    return this.http.get<any>(`${this.baseUrl}/getUsersByGroup/${groupId}`)
  }

  deleteUsersFromGroup(userIds: string[], groupId: string): Observable<boolean> {
    const body = {
      groupId: groupId,
      usersId: userIds 
    };
<<<<<<< HEAD
    return this.http.post<boolean>(`https://localhost:7215/Chat/deleteFromGroup`, body);
=======
    return this.http.post<boolean>(`https://localhost:5208/Chat/deleteFromGroup`, body);
>>>>>>> 875b81d (Initial commit to main)
  }
  
  addUsersToGroup(users: { id: string, recieverName: string, userPhoto: string }[], groupId: string): Observable<any> {
    const body = {
      groupId: groupId,
      users: users
    };

  
    return this.http.post(`${this.baseUrl}/addUserToGroup`, body);
  }
  
}
